src/execnet/gateway_io.py

```diff
@@ -18,6 +18,9 @@
 class Popen2IOMaster(Popen2IO):
     def __init__(self, args, execmodel):
         PIPE = execmodel.subprocess.PIPE
+        if is_frozen_environment():
+            args[0] = str(sys.executable)
```

Comment:
I don't think this is correct, we should not be mutating the input arguments like this, nor assuming the first argument is the interpreter. This class is public, and even if it was not the case, this is certainly surprising behavior.

Clients of this class should instead use `sys.executable` instead of just `"python"`, for example here: https://github.com/AlexPaiva/execnet/blob/227da4174e349b0cc198619b5fd186c6f137df82/src/execnet/gateway_io.py#L86C5-L86C43